### ISSUE TITLE
TS issues in demo

### ISSUE THREAD

**Issue by chrisvxd** [measuredco/puck#537]
## What happened

The following error was shown in VSCode when cloning the project and running the demo using TS 5.5.4.
```
type `UserConfig` does not satisfy the constraint `Config`
```

Screenshot:

![image](https://github.com/user-attachments/assets/0f35b709-18a4-4fc9-b84c-9d21541a74bc)

## What I expected to happen

No error would be shown.

## Steps to reproduce

1. Clone app
2. Open VSCode
3. Check for TS errors

## Notes

* Could be related to mismatched TS versions (4 vs 5)
* Have not tested build
* User reported issue - I have not attempted to reproduce

**Comment by giorgio324** [comment]
I have similar issue, but instead of passing generic to the component I pass it to the Coinfig Type.
This is my config

```
import { DropZone, type Config } from '@measured/puck';

type Components = {
  HeadingBlock: { title: string };
};
type RootProps = {
  backgroundColor: string;
  fullScreen: '100vh' | '275px';
};

type Categories = 'Text';

export const config: Config<Components, RootProps, Categories> = {
  components: {
    HeadingBlock: {
      fields: {
        title: {
          type: 'text',
        },
      },
      render: ({ title }) => {
        return <h1>{title}</h1>;
      },
    },
  },
  root: {
    fields: {
      backgroundColor: {
        type: 'custom',
        render: ({ name, onChange, value }) => {
          console.log(name, onChange, value);
          return <div></div>;
        },
      },
      fullScreen: {
        label: 'Full Screen',
        type: 'radio',
        options: [
          { label: 'Yes', value: '100vh' },
          { label: 'No', value: '275px' },
        ],
      },
    },
    defaultProps: {
      backgroundColor: '#333333',
      fullScreen: '100vh',
    },
    render: ({ backgroundColor, fullScreen }) => {
      return (
        <main
          style={{
            backgroundColor: backgroundColor,
            minHeight: fullScreen,
          }}
        >
          <DropZone zone='body' style={{ minHeight: fullScreen }} />
        </main>
      );
    },
  },
};

export default config;
```
when I import this config and use it in either Puck or Render i get error

Type 'Config<Components, RootProps, "Text">' is not assignable to type 'Config'.
  Types of property 'components' are incompatible.
    Type '{ HeadingBlock: Omit<ComponentConfig<{ title: string; }, { title: string; }, Omit<ComponentData<{ title: string; }>, "type">>, "type">; }' is not assignable to type '{ [x: string]: Omit<ComponentConfig<any, any, Omit<ComponentData<any>, "type">>, "type">; }'.
      Property 'HeadingBlock' is incompatible with index signature.
        Type 'Omit<ComponentConfig<{ title: string; }, { title: string; }, Omit<ComponentData<{ title: string; }>, "type">>, "type">' is not assignable to type 'Omit<ComponentConfig<any, any, Omit<ComponentData<any>, "type">>, "type">'.
          Types of property 'resolveFields' are incompatible.
            Type '((data: Omit<ComponentData<{ title: string; }>, "type">, params: { changed: Partial<Record<"title", boolean>>; fields: Fields<{ title: string; }>; lastFields: Fields<...>; lastData: Omit<...>; appState: AppState; }) => Fields<...> | Promise<...>) | undefined' is not assignable to type '((data: Omit<ComponentData<any>, "type">, params: { changed: Partial<Record<string | number | symbol, boolean>>; fields: Fields<any>; lastFields: Fields<...>; lastData: Omit<...>; appState: AppState; }) => Fields<...> | Promise<...>) | undefined'.ts(2322)
index.d.ts(249, 5): The expected type comes from property 'config' which is declared here on type 'IntrinsicAttributes & { children?: ReactNode; config: Config; data: Partial<Data>; ui?: Partial<UiState> | undefined; ... 10 more ...; dnd?: { ...; } | undefined; }'

I have tryed to upgrade and downgrade typescript version.
For now i use ts ignore to ignore the type issue

**Comment by giorgio324** [comment]
I have managed to get rid of the error by setting typescript strict mode to false in tsconfig.app.json

**Comment by xaviemirmon** [comment]
I've found the issue and release a fix on

PR #545

**Comment by chrisvxd** [comment]
Merged #545 which addresses initial issue.

@giorgio324 I'm unable to reproduce your issue, but please test your config against the latest `@measured/puck@0.16.0-canary.958dc25` and let us know if it is resolved.
